lambdas/gro-publish-record/src/test/java/uk/gov/di/data/lep/PublishRecordTest.java

```diff
@@ -1,27 +1,111 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import uk.gov.di.data.lep.library.config.Config;
 import uk.gov.di.data.lep.library.dto.GroJsonRecord;
 import uk.gov.di.data.lep.library.services.AwsService;
-import uk.gov.di.data.lep.library.services.Mapper;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
 import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.clearInvocations;
 import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.mockConstruction;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
 class PublishRecordTest {
-    private static final PublishRecord underTest = new PublishRecord();
+    private static final AwsService awsService = mock(AwsService.class);
+    private static final Config config = mock(Config.class);
+    private static final HttpClient httpClient = mock(HttpClient.class);
+    private static final HttpResponse<String> httpResponse = mock(HttpResponse.class);
+    private static final ObjectMapper objectMapper = mock(ObjectMapper.class);
+    private static final PublishRecord underTest = new PublishRecord(awsService, config, httpClient, objectMapper);
     private static final Context context = mock(Context.class);
 
+    private static final GroJsonRecord event = new GroJsonRecord("1234567890");
+    private static final HttpRequest expectedAuthRequest = HttpRequest.newBuilder()
+        .uri(URI.create("https://cognitoDomainName.auth.awsRegion.amazoncognito.com/oauth2/token"))
+        .header("Content-Type", "application/x-www-form-urlencoded")
+        .POST(HttpRequest.BodyPublishers.ofString(
+            "grant_type=client_credentials&client_id=cognitoClientId&client_secret=cognitoClientSecret")
+        )        .build();
+    private static final HttpRequest expectedGroRecordRequest = HttpRequest.newBuilder()
+        .uri(URI.create("https://domainName/events/deathNotification"))
+        .header("Authorization", "accessToken")
+        .POST(HttpRequest.BodyPublishers.ofString("{\"sourceId\": \"1234567890\"}"))
+        .build();
+
+    @BeforeAll
+    static void setup() {
+        when(config.getDomainName()).thenReturn("domainName");
+        when(config.getAwsRegion()).thenReturn("awsRegion");
+        when(config.getCognitoClientId()).thenReturn("cognitoClientId");
```

Comment:
as this string, and other strings, is referenced in the above private static initialisation, could we make these values values on the class and then string format them in